Add release v19 handbook (#1778)
giantswarm/docs: src/content/advanced/upgrades/aws-19-release/index.md

---
linkTitle: Giant Swarm Release v19.0.0 for AWS
title: Giant Swarm Release v19.0.0 for AWS
description: AWS release v19 besides many improvements will introduce migration of Cilium as well as replacement of kiam with IRSA. Following handbook should be carefully read by customers upfront the upgrade to prepare the clusters and workloads accordingly.
weight: 10
menu:
  main:
    parent: advanced-upgrades
aliases:
  - /guides/aws-release-v19/
user_questions:
  - Where can I read about v19 changes?
  - What does v19 change?
  - What changes does Cilium bring?
  - What do i have to prepare for v19?
  - Whad do i have to do before IRSA migration?
  - How does the Cilium migration work?
owner:
  - https://github.com/orgs/giantswarm/teams/team-phoenix
last_review_date: 2023-04-13
---

{{< platform_support_table aws="alpha=v19.0.0" >}}

## Introduction

> We have been preparing for a long time to introduce release v19 on AWS. Besides upgrade of components and Kubernetes version to 1.24, this release will involve two major changes for customers i.e. migration from AWS CNI to Cilium and replacement of kiam with IAM Roles for Service Accounts(IRSA).

Reviewer: ```suggestion
We have been preparing for a long time to introduce release v19 on AWS. Besides upgrade of components and Kubernetes version to 1.24, this release will involve two major changes for customers, namely the migration from the AWS VPC CNI to Cilium and the replacement of Kiam with IAM Roles for Service Accounts(IRSA) for authenticating pods against the AWS API.
```

### Key Highlights

- Service Mesh integration: `Cilium` is designed to work seamlessly with popular service meshes like `Istio`, `Linkerd`, and `Envoy`. This allows for more advanced networking and security features, such as mTLS encryption and observability.
- `Cilium` uses a virtual network which provides more flexibility, faster network initialization, and more advanced networking features compared to `AWS CNI` affecting the IP addresses.
- Advanced networking features: `Cilium` supports advanced networking features such as load balancing, network segmentation, and eBPF-based packet filtering. These features allow for more granular control over network traffic and improve security. 
- Scalability: `Cilium's` eBPF-based data plane is highly scalable and performs well even at scale. It is also highly efficient, reducing overhead and maximizing performance.

### What changes with Cilium?

With `Cilium`, you'll no longer be using the `AWS CNI` Pod subnets, so be sure to add custom routes with the `Node subnet(s) CIDR(s)` instead. 

> Additionally, while `Cilium's Network Policy` provides powerful security features, support for setting `ipBlock` with `Pod IPs` is not included, so be sure to inspect your workloads and configured `Network Policies` carefully. The Account Engineers will reach out to you and we will provide a solution to provide the `NetworkCiliumPolicies` before the upgrade for no downtime.

Reviewer: ```suggestion
Additionally, while `Cilium's Network Policy` provides powerful security features, support for setting `ipBlock` with `Pod IPs` is not included, so be sure to inspect your workloads and configured `Network Policies` carefully. The Account Engineers will reach out to you and we will provide a solution to provide the `CiliumNetworkPolicies` before the upgrade for no downtime.
```

> It's important to note that due to changes to `Cluster CR's` during the upgrade process, `GitOps` automation will have to be suspended and any applied changes backported to the repos before resuming. Keep this in mind as you prepare for the upgrade.

Reviewer: ```suggestion
It's important to note that due to changes to `Cluster CR's` during the upgrade process, `GitOps` automation will have to be suspended and any applied changes backported to the repos before resuming. Keep this in mind as you prepare for the upgrade. This needs to be evaluated on a case-by-case basis, since different GitOps implementations might only keep _some_ parts of `Cluster` CRs in Git. Feel free to reach out to your Account Engineer to understand more about these changes.
```

#### Cilium and AWS Load Balancer Controller

If you are running `aws-load-balancer-controller` inside your clusters for managing [Network Load Balancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html) and you did set the annotation `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip`, you need to change it to `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance`.

For further information, please checkout the [documentation](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#traffic-routing)

## IAM roles for service accounts (IRSA)

By switching from `KIAM` to `IAM Roles for Service Accounts (IRSA)`, we're making it easier and more secure for your Kubernetes workloads to interact with AWS services. 

### Key Highlights

- Official AWS way to authenticate pods to AWS API.
- Reduced complexity: IRSA eliminates the need for a separate service like KIAM, streamlining your Kubernetes clusters.
- Regional STS (Security Token Service) rather than using global STS

> ### What changes with IRSA?

Reviewer: ```suggestion

### What changes with IRSA?
```

- `Cilium` uses a virtual network which provides more flexibility, faster network initialization, and more advanced networking features compared to `AWS CNI` affecting the IP addresses.
- Advanced networking features: `Cilium` supports advanced networking features such as load balancing, network segmentation, and eBPF-based packet filtering. These features allow for more granular control over network traffic and improve security. 
- Scalability: `Cilium's` eBPF-based data plane is highly scalable and performs well even at scale. It is also highly efficient, reducing overhead and maximizing performance.

### What changes with Cilium?

With `Cilium`, you'll no longer be using the `AWS CNI` Pod subnets, so be sure to add custom routes with the `Node subnet(s) CIDR(s)` instead. 

Additionally, while `Cilium's Network Policy` provides powerful security features, support for setting `ipBlock` with `Pod IPs` is not included, so be sure to inspect your workloads and configured `Network Policies` carefully. The Account Engineers will reach out to you and we will provide a solution to provide the `NetworkCiliumPolicies` before the upgrade for no downtime.

> It's important to note that due to changes to `Cluster CR's` during the upgrade process, `GitOps` automation will have to be suspended and any applied changes backported to the repos before resuming. Keep this in mind as you prepare for the upgrade

Reviewer 2: ```suggestion
It's important to note that due to changes to `Cluster CR's` during the upgrade process, `GitOps` automation will have to be suspended and any applied changes backported to the repos before resuming. Keep this in mind as you prepare for the upgrade.
```

#### Cilium and AWS Load Balancer Controller

If you are running `aws-load-balancer-controller` inside your clusters for managing [Network Load Balancer](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/introduction.html) and you did set the annotation `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip`, you need to change it to `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance`.

For further information, please checkout the [documentation](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#traffic-routing)

## IAM roles for service accounts (IRSA)

By switching from `KIAM` to `IAM Roles for Service Accounts (IRSA)`, we're making it easier and more secure for your Kubernetes workloads to interact with AWS services. 

### Key Highlights

- Official AWS way to authenticate pods to AWS API.
- Reduced complexity: IRSA eliminates the need for a separate service like KIAM, streamlining your Kubernetes clusters.
- Regional STS (Security Token Service) rather than using global STS

> 
> 

Reviewer 2: ```suggestion

```
